notebooks/working/3.0-mab-independent-variable-simulation.md

```diff
@@ -195,185 +197,551 @@ class FitDistribution(object):
+                # Use the Fitter library to fit distributions
+                # to the data
+                fitter_object = Fitter(data=trip_var,
+                                       distributions=cont_dists,
+                                       timeout=30)
+                fitter_object.fit()
+                # Get the best distribution and store in dictionary
+                BestDict = fitter_object.get_best()
+                params_dict[var]['distribution'] = list(BestDict.items())[0][0]
+                params_dict[var]['parameters'] = list(BestDict.items())[0][1]
+
+    return params_dict
+
+
+def SimDf(params_dict, size=1000):
+    """Funtion to simulate data of size N based on specified
+    distribution/parameters found by the fitter package.
+    Inputs:
+    -------
+    data: dataframe from which columns are to be taken
+    dist_params: the distribution parameters from find_dist_df
+    Outputs:
+    -------
+    DataFrame object with simulated data based on specified distributions
+    """
+    Sim_Df = pd.DataFrame(columns=list(params_dict.keys()))
+    Sim_Df = Sim_Df.fillna(0)
+    for column in list(params_dict.keys()):
+        if params_dict[column]['distribution'] == 'categorical':
+            data_sim = np.random.choice(a=params_dict[column]['parameters'][0],
+                                        p=params_dict[column]['parameters'][1],
+                                        size=size)
+            Sim_Df[column] = data_sim
+        elif params_dict[column]['distribution'] == 'constant':
+            data_sim = params_dict[column]['parameters'][0]
+            Sim_Df[column] = data_sim
+        else:
+            dist = getattr(scipy.stats, params_dict[column]['distribution'])
+            data_sim = dist.rvs(*params_dict[column]['parameters'], size=size)
+            Sim_Df[column] = data_sim
+    return Sim_Df
+
+
+def SimulateAvailability(data_long, sim_data, obs_id_col, alt_name_dict):
+
+    series = pd.Series([])
+    for i, obs in zip(np.arange(len(data_long[obs_id_col].unique())), data_long[obs_id_col].unique()):
+        series[i] = data_long[data_long[obs_id_col] == obs].shape[0]
+
+    av_size = sim_data.shape[0]
+    alts_sim = np.random.choice(a=np.arange(series.max()+1),
+                                p=np.bincount(series)/len(series),
+                                size=av_size)
+
+    N = len(alt_name_dict)
+
+    av_sim = [np.array([1] * K + [0]*(N-K)) for K in alts_sim]
+
+    for x in av_sim:
+        np.random.shuffle(x)
+
+    np.random.shuffle(av_sim)
+    AV_columns = [alt_name_dict[i]+'_AV' for i in alt_name_dict.keys()]
+    AV_Df = pd.DataFrame(av_sim, columns=AV_columns)
+    choice = [random.choice(np.nonzero(a == 1)[0]) + 1 for a in np.array(AV_Df)]
+    choice_df = pd.DataFrame(choice, columns=['sim_choice'])
+    Sim_DF_AV = pd.concat([sim_data, AV_Df, choice_df], axis=1, sort=False)
+    return Sim_DF_AV
+```
+
+# Functions for simulating choices - to be imported as library
```

Comment:
```suggestion
# Functions for simulating choices - to be imported as library
import sys
sys.path.append('../../src')
from viz.utils import (progress, 
                       simulate_choice_vector)
```

Reply:
This will perform the imports I think you want without needing to copy the functions into your notebook.